### PR TITLE
Support JarJar

### LINKED ISSUE #92: Support Forge Jar-In-Jar (JarJar)

### PR DESCRIPTION
**Pull request by shedaniel**
Close #92

We are waiting of https://github.com/MinecraftForge/MinecraftForge/pull/8822

This is more changes than I would have liked, but the system requires the maven notation and the version.
The version range is generated from `[VERSION,)`, which is VERSION or above. This should be very straight forward.

Adding a dependency for nesting has the same syntax as Fabric:
```groovy
include "group:module:version"
```

Signed-off-by: shedaniel <daniel@shedaniel.me>

### CODE REVIEW

**Comment by shedaniel** [comment]
We are now waiting for https://github.com/MinecraftForge/MinecraftForge/issues/8843

**Comment by ThatGravyBoat** [comment]
Any update on this as the issue in forge has been fixed.

File: src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java [hunk 1]
```diff
@@ -67,16 +73,34 @@ public Provider<ConfigurableFileCollection> getNestedJars(final Configuration co
...
+		return project.provider(() -> {
+			final List<Pair<Metadata, RegularFileProperty>> files = new ArrayList<>();
+			final Set<String> visited = Sets.newHashSet();
+			final Set<Task> builtBy = Sets.newHashSet();
+
+			files.addAll(getProjectDeps(configuration, visited, builtBy));
+			files.addAll(getFileDeps(configuration, visited, builtBy));
+			return new Pair<>(files, task -> {
+				TaskDependency dependencies = configuration.getBuildDependencies();
+				Set<Task> tasks = (Set<Task>) dependencies.getDependencies(task);
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/build/nesting/IncludedJarFactory.java]: Should be `new HashSet<>(...)`

File: src/main/java/net/fabricmc/loom/build/nesting/JarNester.java [hunk 2]
```diff
@@ -132,4 +139,42 @@ public static void nestJars(Collection<File> jars, File modJar, ModPlatform plat
 			throw new java.io.UncheckedIOException("Failed to nest jars into " + modJar.getName(), e);
 		}
 	}
+
+	private static void handleForgeJarJar(Collection<File> jars, List<Pair<IncludedJarFactory.Metadata, File>> forgeJars, File modJar, Logger logger) throws IOException {
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/build/nesting/JarNester.java]: Probably shouldn't take the plain jars collection, doesn't seem to be used

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**